src/main/java/org/wso2/carbon/identity/conditional/auth/typingdna/VerifyUserWithTypingDNAFunctionImpl.java

```diff
@@ -0,0 +1,206 @@
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.JsGraphBuilder;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticationContext;
+import org.wso2.carbon.identity.conditional.auth.functions.common.utils.CommonUtils;
+import org.wso2.carbon.identity.conditional.auth.typingdna.exception.TypingDNAAuthenticatorException;
+import org.wso2.carbon.identity.event.IdentityEventException;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.net.ssl.HttpsURLConnection;
+
+import static org.wso2.carbon.identity.conditional.auth.functions.common.utils.Constants.OUTCOME_FAIL;
+import static org.wso2.carbon.identity.conditional.auth.functions.common.utils.Constants.OUTCOME_SUCCESS;
+
+/**
+ * Custom adaptive function implementation for verifying
+ * users typing pattern with TypingDNA.
+ */
+public class VerifyUserWithTypingDNAFunctionImpl implements VerifyUserWithTypingDNAFunction {
+
+    private static final Log log = LogFactory.getLog(VerifyUserWithTypingDNAFunctionImpl.class);
+
+    /**
+     * Function to send verify request to typingDNA APIs.
+     *
+     * @param context       Context from authentication flow.
+     * @param eventHandlers
+     * @throws TypingDNAAuthenticatorException
+     */
+    @Override
+    public void verifyUserWithTypingDNA(JsAuthenticationContext context, Map<String, Object> eventHandlers) throws TypingDNAAuthenticatorException {
+
+        try {
+            JsAuthenticatedUser user = Utils.getUser(context);
+            String username = user.getWrapped().getUserName();
+            String tenantDomain = user.getWrapped().getTenantDomain();
+            String typingPattern = Utils.getTypingPattern(context);
+
+            AtomicBoolean result = new AtomicBoolean(false);
+
+            // Getting connector configurations.
+            String APIKey = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.USERNAME, tenantDomain);
+            String APISecret = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.CREDENTIAL, tenantDomain);
+            String advanced = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.ADVANCE_MODE_ENABLED, tenantDomain);
+            String region = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.REGION, tenantDomain);
+            String Enabled = CommonUtils.getConnectorConfig(TypingDNAConfigImpl.ENABLE, tenantDomain);
```

Comment:
Shall we use boolean here as well. You can use Boolean.parseBoolean(). Change the usages accordingly.